src/handlers/gestures/GestureDetector/index.tsx

```diff
@@ -0,0 +1,271 @@
+      )
+    );
+  }
+
+  function onHandlersUpdate(skipConfigUpdate?: boolean) {
+    // if the underlying view has changed we need to reattach handlers to the new view
+    const viewTag = findNodeHandle(state.viewRef) as number;
+    const forceReattach = viewTag !== state.previousViewTag;
+
+    if (forceReattach || needsToReattach(preparedGesture, gesturesToAttach)) {
+      validateDetectorChildren(state.viewRef);
+      dropHandlers(preparedGesture);
+      attachHandlers({
+        preparedGesture,
+        gestureConfig,
+        gestures: gesturesToAttach,
+        webEventHandlersRef,
+        viewTag,
+        mountedRef,
+      });
+
+      state.previousViewTag = viewTag;
+      state.forceReattach = forceReattach;
+      if (forceReattach) {
+        forceRender();
+      }
+    } else if (!skipConfigUpdate) {
+      updateHandlers(
+        preparedGesture,
+        gestureConfig,
+        gesturesToAttach,
+        mountedRef
+      );
+    }
+  }
+
+  // Reanimated event should be rebuilt only when gestures are reattached, otherwise
+  // config update will be enough as all necessary items are stored in shared values anyway
+  const needsToRebuildReanimatedEvent =
+    state.firstRender ||
+    needsToReattach(preparedGesture, gesturesToAttach) ||
+    state.forceReattach;
+
+  state.forceReattach = false;
+
+  if (shouldUseReanimated) {
+    // Whether animatedGesture or gesture is used shouldn't change while the app is running
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useAnimatedGesture(preparedGesture, needsToRebuildReanimatedEvent);
+  }
+
+  useEffect(() => {
+    const viewTag = findNodeHandle(state.viewRef) as number;
+    mountedRef.current = true;
+
+    validateDetectorChildren(state.viewRef);
+
+    attachHandlers({
+      preparedGesture,
+      gestureConfig,
+      gestures: gesturesToAttach,
+      webEventHandlersRef,
+      viewTag,
+      mountedRef,
+    });
+
+    return () => {
+      mountedRef.current = false;
+      dropHandlers(preparedGesture);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!state.firstRender) {
+      onHandlersUpdate();
+    } else {
+      state.firstRender = false;
+    }
+  }, [props]);
+
+  const refFunction = (ref: unknown) => {
+    if (ref !== null) {
+      // @ts-ignore Just setting the view ref
+      state.viewRef = ref;
+
+      // if it's the first render, also set the previousViewTag to prevent reattaching gestures when not needed
+      if (state.previousViewTag === -1) {
+        state.previousViewTag = findNodeHandle(state.viewRef) as number;
+      }
+
+      // pass true as `skipConfigUpdate`, here we only want to trigger the eventual reattaching of handlers
+      // in case the view has changed, while config update would be handled be the `useEffect` above
+      onHandlersUpdate(true);
+
+      if (__DEV__ && isFabric() && global.isFormsStackingContext) {
```

Comment:
```suggestion
      if (!__DEV__ || !isFabric() || !global.isFormsStackingContext) {
        return;
      }
```

Reply:
This one feels like too much of a microoptimization 😅. For my small brain, it's much easier to parse a conjunction than negated alternatives.

Reply:
I get it, let's leave that as it is now.